src/memory-analyzer/analyze_symbol.cpp

```diff
@@ -0,0 +1,375 @@
+/*******************************************************************\
+
+Module: Symbol Analyzer
+
+Author: Malte Mues <mail.mues@gmail.com>
+        Daniel Poetzl
+
+\*******************************************************************/
+
+#include "analyze_symbol.h"
+
+#include <util/c_types.h>
+#include <util/c_types_util.h>
+#include <util/config.h>
+#include <util/expr_initializer.h>
+#include <util/string_constant.h>
+
+gdb_value_extractort::gdb_value_extractort(
+  const symbol_tablet &symbol_table,
+  const char *binary)
+  : gdb_api(binary),
+    symbol_table(symbol_table),
+    ns(symbol_table),
+    c_converter(ns, expr2c_configurationt::clean_configuration),
+    allocate_objects(ID_C, source_locationt(), "", this->symbol_table)
+{
+}
+
+void gdb_value_extractort::analyze_symbols(
+  const std::vector<std::string> &symbols)
```

Comment:
I think this should be a `std::vector<irep_idt>` to avoid repeated `std::string` -> `irep_idt` conversion.

Reply:
No problem, but the way we get the vector is by calling `split_string`. I generalised it to take `irep_idt`s as well. Now it seems to be a bit bigger change that originally intended (https://github.com/danpoe/cbmc/pull/9/commits/38b9cb1d1d57ba08589ddb89e787a38ab26dc3f0)

Reply:
It was suggested that `split_string` actually shouldn't work for `irep_idt`s. I changed the fix and now we transform the vector of strings (from `split_string`) to a vector of `irep_idt` before passing it to `analyze_symbols`. Is this solution ok https://github.com/danpoe/cbmc/pull/9/commits/b4555e55e4f71fe88e4c7560c146b30a2cebcc8e?